engines/mm/detection_tables.h

```diff
@@ -19,9 +19,39 @@
  *
  */
 
-namespace Xeen {
+namespace MM {
+
+static const MightAndMagicGameDescription GAME_DESCRIPTIONS[] = {
+	{
+		// Might and Magic 1
+		{
+			"mm1",
+			nullptr,
+			AD_ENTRY1s("wallpix.dta", "86a7ef03fd5bf434d83012b07fa92680", 123059),
+			Common::EN_ANY,
+			Common::kPlatformDOS,
+			ADGF_UNSTABLE,
+			GUIO0()
+		},
+		GType_MightAndMagic1,
+		0
+	},
+
+	{
+		// Might and Magic 1 (Enhanced)
+		{
+			"mm1_enh",
+			nullptr,
+			AD_ENTRY1s("wallpix.dta", "86a7ef03fd5bf434d83012b07fa92680", 123059),
+			Common::EN_ANY,
+			Common::kPlatformDOS,
+			ADGF_UNSTABLE,
+			GUIO0()
+		},
+		GType_MightAndMagic1,
+		GF_ENHANCED
```

Comment:
In general, I'd recommend removing this field and move the flags to the ADGameDescription::flags. E.g. write `GF_ENHANCED | ADGF_UNSTABLE`.

I do not understand why people keep putting flags separately :D

Reply:
I wonder if it would be worth changing all engines to use a separate field, so that we can either reduce the size of the field or use all 32 bits for generic AD flags?

Reply:
Wouldn't that, on the contrary, increase the size of the structure?

If you switch to `uint16` for the flags, two things: (1) we are already using 16, and we're close to overflowing to 17 bits (2) structs are aligned to 32 or 64 bits. Thus, in fact, you will not save memory by switching to 16 bits, and if you consider packing it, then that is a performance hit.

Reply:
> Wouldn't that, on the contrary, increase the size of the structure?

Not all engines (IIRC about 10-20 out of almost 100) store custom flags in this field, and in cases like this one, making both `gameID` and `features` 8-bit values would cancel out the size increase for the detection entries in this engine.